### ISSUE TITLE
problem with dfki-ot-hsmm voice (tr locale)

### ISSUE THREAD

**Issue by yilmazay74** [marytts/marytts#905]
Hi,
I am new to MaryTTs, I wanted to run the demo from marytts github.
It worked with the default English voice cmu-slt-hsmm.
However, I want it to work with Turkish voice dfki-ot-hsmm voice.
No matter what I do, I cannot make it work.
Everything seems in place.
Could anyone help me with this?
I am attaching a screenshot of my env and the error messages.
From left side you can see that all necessary jars are added in the project.

Thanks in Advance
Best REgards
Yılmaz A.
![marytts error](https://user-images.githubusercontent.com/17269740/49635781-df260900-fa11-11e8-96aa-cc60fc4da3b0.PNG)

**Comment by psibre** [comment]
Can you paste your `pom.xml`?

**Comment by yilmazay74** [comment]
[pom.xml.txt](https://github.com/marytts/marytts/files/2660372/pom.xml.txt)
Hi Psibre,
I attached my pom.xml. Please let me know what is missing in the pom
Thanks in advance.

REgards

**Comment by psibre** [comment]
I think you need to get rid of the system scope. It's preventing you from resolving the transitive dependencies.

**Comment by yilmazay74** [comment]
[MavenBuild Error after removing system path dependencies in pom.xml.txt](https://github.com/marytts/marytts/files/2660416/MavenBuild.Error.after.removing.system.path.dependencies.in.pom.xml.txt)
Hi, 
When I remove system scopes in the pom.xml, then maven cannot find marytts-builder and lexicon dependencies. I attached the maven build error stack trace.

**Comment by psibre** [comment]
I'd stay away from jitpack. You should resolve the MaryTTS dependencies from JCenter.

**Comment by yilmazay74** [comment]
Ok, I replaced it with JCenter, and then it can't find some other jars, so I kept both repos, then it successfully compiled. However, it still gives the same error message 👍 Locale: en_GB
marytts.exceptions.SynthesisException: cannot process
	at marytts.LocalMaryInterface.process(LocalMaryInterface.java:440)
	at marytts.LocalMaryInterface.generateAudio(LocalMaryInterface.java:348)
	at marytts.TextToSpeech.speak(TextToSpeech.java:65)
	at Try_MarryTTS_IF_Working.MaryTTSTester.lambda$main$3(MaryTTSTester.java:99)
	at java.util.Arrays$ArrayList.forEach(Arrays.java:3880)
	at Try_MarryTTS_IF_Working.MaryTTSTester.main(MaryTTSTester.java:99)
Caused by: java.lang.UnsupportedOperationException: No known way of generating output (AUDIO) from input(RAWMARYXML), no processing path through modules.

By the way, when I compile the project from marytts github, and compile and run, it gives the same error
Whatever I try, it stucks at this location. 
I started to think whether there is a problem with that voice dfki-ot-hsmm or not.
Is there anyone who had successfully made this work?

**Comment by psibre** [comment]
> By the way, when I compile the project from marytts github, and compile and run, it gives the same error

Could you please elaborate? What command are you running, and what is the error?

**Comment by yilmazay74** [comment]
Actually, what I am doing is very simple. I cloned the marytts code from this site. And then txt2wav maven sample project. It works with default en_US locale and cmu-slt-hsmm voice.
Then I added Turkish locale (tr) and dfki-ot-hsmm voice which is provided by dfki.com.de.
I added necessary dependencies in the pom.xml.
Below is the available voice and locale list. As it can be seen Turkish locale and dfki-ot-hsmm is among the available locales and voices.
Avaliable Locales:
Locale: tr
Locale: en_US
Locale: en_GB

Avaliable Voices:
Voice: dfki-ot-hsmm
Voice: cmu-slt-hsmm

So, when I run the demo MaryTTSTester in the code below:

		String text = "Merhaba dostum";
		TextToSpeech tts = new TextToSpeech();
	        tts.setLocale(new Locale("tr"));
                tts.setVoice("dfki-ot-hsmm");
	 tts.speak(text , 2.0f, false, true));
----------------------------------------
related code in TextToSpeech: 
public void speak(String text , float gainValue , boolean daemon , boolean join) {
		
		// Stop the previous player
		stopSpeaking();
		
		try (AudioInputStream audio = marytts.generateAudio(text)) {
			
			// Player is a thread(threads can only run one time) so it can be
			// used has to be initiated every time
			tts = new AudioPlayer();
			tts.setAudio(audio);
			tts.setGain(gainValue);
			tts.setDaemon(daemon);
			tts.start();
			if (join)
				tts.join();
			
		} catch (SynthesisException ex) {
			Logger.getLogger(getClass().getName()).log(Level.ALL, "Error saying phrase.", ex);
			ex.printStackTrace();
		} catch (IOException ex) {
			Logger.getLogger(getClass().getName()).log(Level.ALL, "IO Exception", ex);
		} catch (InterruptedException ex) {
			Logger.getLogger(getClass().getName()).log(Level.ALL, "Interrupted ", ex);
			tts.interrupt();
		}
	}
It throws error at try (AudioInputStream audio = marytts.generateAudio(text)) {  line
and caught by SynthesisException.
The error message is:
marytts.exceptions.SynthesisException: cannot process
	at marytts.LocalMaryInterface.process(LocalMaryInterface.java:440)
	at marytts.LocalMaryInterface.generateAudio(LocalMaryInterface.java:348)
	at marytts.TextToSpeech.speak(TextToSpeech.java:65)
	at Try_MarryTTS_IF_Working.MaryTTSTester.lambda$3(MaryTTSTester.java:101)
	at java.util.Arrays$ArrayList.forEach(Arrays.java:3880)
	at Try_MarryTTS_IF_Working.MaryTTSTester.main(MaryTTSTester.java:101)
Caused by: java.lang.UnsupportedOperationException: No known way of generating output (AUDIO) from input(RAWMARYXML), no processing path through modules.
	at marytts.server.Request.processOneChunk(Request.java:524)
	at marytts.server.Request.processOrLookupOneChunk(Request.java:403)
	at marytts.server.Request.process(Request.java:337)
	at marytts.LocalMaryInterface.process(LocalMaryInterface.java:438)

**Comment by psibre** [comment]
You code samples don't show how you're using your MaryInterface instance `marytts`. But the stacktrace indicates you might have set the wrong input type. For `generateAudio(String text)` it must be `MaryDataType.TEXT`.

**Comment by psibre** [comment]
Btw, here is a MWE using Gradle

### `build.gradle`
```groovy
plugins {
    id 'application'
}

repositories {
    jcenter()
}

dependencies {
    compile group: 'de.dfki.mary', name: 'voice-dfki-ot-hsmm', version: '5.2'
}

mainClassName = 'HelloTurkishMaryTTS'
```
### `src/main/java/HelloTurkishMaryTTS.java`
```java
import marytts.LocalMaryInterface;

public class HelloTurkishMaryTTS {

    public static void main(String[] args) throws Exception {
        LocalMaryInterface marytts = new LocalMaryInterface();
        marytts.generateAudio("Merhaba dostum");
    }
}
```
I suggest you start with that, and add your complexity, taking note when it stops working.

**Comment by yilmazay74** [comment]
Hi psibre,
The build is done successfully without problems. 
However, when I run the application I am getting errors.
Basically, It cannot start LocalMaryServer instance.
I am getting the following error messages:
-------------------------------
Exception in thread "main" marytts.exceptions.MaryConfigurationException: Cannot start MARY server
	at marytts.LocalMaryInterface.<init>(LocalMaryInterface.java:66)
	at de.dfki.mary.HelloTurkishMaryTTS.main(HelloTurkishMaryTTS.java:7)
Caused by: java.lang.Exception: Problem starting module Synthesis
	at marytts.server.Mary.startModules(Mary.java:151)
	at marytts.server.Mary.startup(Mary.java:297)
	at marytts.server.Mary.startup(Mary.java:204)
	at marytts.util.MaryRuntimeUtils.ensureMaryStarted(MaryRuntimeUtils.java:72)
	at marytts.LocalMaryInterface.<init>(LocalMaryInterface.java:64)
	... 1 more
Caused by: marytts.exceptions.MaryConfigurationException: Cannot instantiate voice 'dfki-ot-hsmm'
	at marytts.modules.synthesis.Voice.<init>(Voice.java:184)
	at marytts.htsengine.HMMVoice.<init>(HMMVoice.java:94)
	at marytts.modules.synthesis.HMMSynthesizer.startup(HMMSynthesizer.java:157)
	at marytts.modules.Synthesis.startupSynthesizers(Synthesis.java:73)
	at marytts.modules.Synthesis.startup(Synthesis.java:65)
	at marytts.server.Mary.startModules(Mary.java:149)
	... 5 more
Caused by: marytts.exceptions.MaryConfigurationException: No allophone set specified -- neither for voice 'dfki-ot-hsmm' nor for locale 'tr'
	at marytts.modules.synthesis.Voice.init(Voice.java:207)
	at marytts.modules.synthesis.Voice.<init>(Voice.java:182)
	... 10 more
Caused by: marytts.exceptions.MaryConfigurationException: No such property: tr.allophoneset
	at marytts.util.MaryRuntimeUtils.needAllophoneSet(MaryRuntimeUtils.java:315)
	at marytts.modules.synthesis.Voice.init(Voice.java:205)
	... 11 more
Exception in thread "Thread-1" java.lang.IllegalStateException: MARY system is not running
	at marytts.server.Mary.shutdown(Mary.java:371)
	at marytts.server.Mary$2.run(Mary.java:290)
-----------------------------------

**Comment by yilmazay74** [comment]
Hi psibre,
I attached my sample project. Could you please have a look at it and let me know what is wrong with the project.?
I appreciate your help.
Best Regards
Yılmaz A.

[marytts-txt2wav.zip](https://github.com/marytts/marytts/files/2670973/marytts-txt2wav.zip)

**Comment by psibre** [comment]
@yilmazay74 If you're basing your sample project on [marytts-txt2wav](/marytts/marytts-txt2wav), why don't you fork it, instead of attaching a zip?

**Comment by yilmazay74** [comment]
Hi Psibre,
I just want to make Turkish locale work, but I could not make it work yet. 
Even the sample code you gave me threw error. Ok, there was not much meaning to attach the zip, I accept that.
However, I would like to know if there was a missing dependency or a configuration mistake in the project.
Anyways, I deleted my previous comment.
I would appreciate if you could let me know, why LocalMAryServer is throwing error in your sample code that you gave me?
Best Regards

**Comment by psibre** [comment]
> Even the sample code you gave me threw error.

Then let's start with that! What was the error?

**Comment by psibre** [comment]
Which command did you run?

**Comment by yilmazay74** [comment]
![maryttsserverrunerrorscreenshot](https://user-images.githubusercontent.com/17269740/49876888-51915180-fe35-11e8-92a0-8f7332081258.png)
HelloTurkishMaryTTS > Run AS > Java Application

**Comment by yilmazay74** [comment]
![maryttsserverrunerrorscreenshot2](https://user-images.githubusercontent.com/17269740/49877127-c5cbf500-fe35-11e8-9220-64908a47745f.png)

**Comment by psibre** [comment]
Oh, never mind. I just spotted the mistake in my code -- I'd forgotten to set the locale! =( Sorry about the confusion...

For good measure, here is a slightly extended version of the main class that also saves the audio to a wav file for inspection:

### src/main/java/HelloTurkishMaryTTS.java

```java
import marytts.LocalMaryInterface;
import marytts.util.data.audio.MaryAudioUtils;

import javax.sound.sampled.AudioInputStream;
import java.util.Locale;

public class HelloTurkishMaryTTS {

    public static void main(String[] args) throws Exception {
        LocalMaryInterface marytts = new LocalMaryInterface();
        marytts.setLocale(Locale.forLanguageTag("tr"));
        AudioInputStream audio = marytts.generateAudio("Merhaba dostum");
        MaryAudioUtils.writeWavFile(MaryAudioUtils.getSamplesAsDoubleArray(audio), "Merhaba.wav", audio.getFormat());
    }
}
```

**Comment by psibre** [comment]
On the command line, run the command:
```
$ gradle run
```
Making sure to use a version of Gradle *lower* than the latest (but currently incompatible) 5.0.

**Comment by yilmazay74** [comment]
[MaryTTS Gradle Run Error Log.txt](https://github.com/marytts/marytts/files/2672310/MaryTTS.Gradle.Run.Error.Log.txt)

**Comment by yilmazay74** [comment]
My gradle version is 4.10.2

**Comment by psibre** [comment]
OK, here's what it looks like for me:
![test](https://user-images.githubusercontent.com/1278639/49883280-9ff12100-fe32-11e8-9d6d-1a6a4e52baaf.gif)
I can see that you're trying this on Windows, which is arguably not an ideal platform for development.
I don't even have access to a Windows machine, so I can't reproduce your issues.
But it works on OSX and Linux without problems.

**Comment by yilmazay74** [comment]
Hi psibre,
Thanks for your comments and support. Finally, I deleted all files under maven's local repository (.m2) under de.dfki folder, and then I rebuilt the project, and it worked. Apparantly, Eclipse maven has some problems. Although I always use  maven clean install, it seems that it can not clean everthing and some jars maybe conflicting. Whatever it is, by manually deleting .m2/de/dfki/ solved my problem.
Thanks a lot.
Best Regards
